pkg/config/get.go

```diff
@@ -146,3 +150,40 @@ func GetGHReleases(ctx context.Context, client *github.Client, owner, repo strin
 
 	return allReleases, nil
 }
+
+func GetAppDefaultsConfig(content []byte, subKey, appName string) (*model.AppDefaultsConfig, error) {
+	var (
+		data             map[string]interface{}
+		allConfigs       model.AppDefaultsConfig
+		availableConfigs model.AppDefaultsConfig
+	)
+
+	if subKey == "" {
+		if err := yaml.Unmarshal(content, &allConfigs); err != nil {
+			return nil, err
+		}
+	} else {
+		if err := yaml.Unmarshal(content, &data); err != nil {
+			return nil, err
+		}
+		data, ok := data[subKey].(map[string]interface{})
```

Comment:
Avoid variable shadowing so we can make the error message more descriptive.
```suggestion
		subData, ok := data[subKey].(map[string]interface{})
```